### DISCUSSION TITLE
Javascript execute multiple times

### DISCUSSION THREAD

**Discussion by ryzam** [Budibase/budibase#3955] [Help] [answered]
Hi guys,

In my javascript function I just add simple log console when update state for debugging purpose but I realize when execute the function budibase will call multiple times even the define action only call one Update State

![image](https://user-images.githubusercontent.com/597759/148494627-da3786c4-c5d3-49a7-ad75-94660131013a.png)

![image](https://user-images.githubusercontent.com/597759/148494739-792e26f3-395f-4f2d-830b-d54472613f9a.png)

![image](https://user-images.githubusercontent.com/597759/148494893-8fa6dc39-bff5-4f68-a6da-0e2b42bf9206.png)

**Comment by mjashanks** [comment, marked as answer]
Thanks for reporting @ryzam, I have logged this as an issue: https://github.com/Budibase/budibase/issues/3964

**Reply by mjashanks** [reply]
@ryzam checkout reply: https://github.com/Budibase/budibase/issues/3964

Basically, binding functions should not (cannot) be used to make any state changes. They should only return a value. Therefore it shouldn't matter if they get executed multiple times.

But thanks for flagging this anyway, will be useful for others in the future.
